File: DevelopArchitecture/DevelopArchitecture/architecture/baseObject/JTableDataArrModel.m
//
//  JTableDataArrModel.m
//  DevelopArchitecture
//
//  Created by Jeffrey on 15/7/21.
//  Copyright (c) 2015年 Jeffrey. All rights reserved.
//

#import "JTableDataArrModel.h"
@implementation JTableDataArrModel

maddProperyValue(m_titleArr, NSMutableArray)
maddProperyValue(m_dataModelArr, NSMutableArray)
maddProperyValue(m_dataTypeArr, NSMutableArray)
maddProperyValue(m_otherMarkArr, NSMutableArray)

-(void)funj_setTableShowTypeForSection:(NSInteger)section o:(BOOL)isOpen{
    if(section>=[_m_dataTypeArr count]){
        [_m_dataTypeArr addObject:isOpen?@"YES":@"NO" ];
    }else{
        _m_dataTypeArr[section]=isOpen?@"YES":@"NO";
    }
}
-(NSArray*)funj_getShowDataWithSection:(NSInteger)section{
    if(section<_m_dataTypeArr.count && [_m_dataTypeArr[section]isEqualToString:@"YES"]){
        return _m_dataModelArr[section];
    }else{
        return @[];
    }
}
-(BOOL)funj_getDataTypeWithSection:(NSInteger)section{
    if([_m_dataTypeArr[section]isEqualToString:@"YES"])return YES;
    else return NO;
}

@end

File: DeveloperArchitecture/DeveloperArchitecture/architecture/baseObject/JTableDataArrModel.swift
//
//  JTableDataArrModel.swift
//  DeveloperArchitecture
//
//  Created by Jeffrey on 2020/10/30.
//  Copyright © 2020 Jeffery. All rights reserved.
//

import Foundation

class JTableDataArrModel : JBaseDataModel {
    lazy var m_titleArr: [Any] = { return [] }()
    lazy var m_dataModelArr: [Any] = { return [] }()
    lazy var m_dataTypeArr: [String] = { return [] }()
    lazy var m_otherMarkArr: [Any] = { return [] }()
    
    func funj_setTypeForSection(_ section : Int , isOpen : Bool) {
        if section >= m_dataTypeArr.count {
            m_dataTypeArr.append(isOpen ? "1" : "0")
        } else {
            m_dataTypeArr[section] = isOpen ? "1" : "0"
        }
    }
    func funj_getDataWithSecton(_ section : Int) ->[Any] {
        if section < m_dataTypeArr.count && m_dataTypeArr[section] == "1" {
            return m_dataModelArr[section] as! [Any]
        } else {
            return []
        }
    }
    func funj_getTypeWithSection(_ section : Int) -> Bool {
        if m_dataTypeArr[section] == "1" { return true}
        return false
    }
}
